### PR TITLE
refactor: rework test setup

### PR DESCRIPTION
**Pull request by weichweich**
## fixes https://github.com/KILTprotocol/ticket/issues/2374

Move all code examples into a single project.

* quicker installs
* share code & helpers
* same code style everywhere

## how to test

```bash
# test everything
yarn test

# test the individual sections
yarn test workshop
yarn test dapp
yarn test core
yarn test staking
```

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/principles/boy-scout-rule)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: code_examples/sdk_examples/src/test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  program
+    .command('all', { isDefault: true })
+    .description('Run all tests')
+    .action(() => {
+      for (const key in whichToRun) {
+        whichToRun[key] = true
+      }
+    })
+  program
+    .command('workshop')
```
Comment (rflechtner) [inline code_examples/sdk_examples/src/test.ts:29]: how about `section` subcommand with a positional argument with limited choices? E.g.
```js
.addArgument(new Argument('section', 'only run tests for selected section').argRequired().choices(['workshop', 'dapp', 'core', 'staking']))

```
Or if positional don't work well with commander, you could add a mandatory `which` arg
```js
.addOption(new Option('-w, --which <section>', 'select section for which to run tests').choices(['workshop', 'dapp', 'core', 'staking']).makeOptionMandatory(true))
```

File: code_examples/sdk_examples/src/test.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  const wssAddress =
+    process.env.WSS_ADDRESS || 'wss://peregrine.kilt.io/parachain-public-ws'
+  const faucetSeed = process.env[SEED_ENV]
+
+  if (!faucetSeed) {
+    console.log(
+      `Account seed with sufficient balance is required. Set the secret seed using the ${SEED_ENV} environment variable.`
+    )
+    throw 'Account seed is missing'
```
Comment (rflechtner) [inline code_examples/sdk_examples/src/test.ts]: probably doesn't matter, but you are throwing a string here (yes, you can do that in js). If you want to throw an error, you have to be explicit
```suggestion
    throw new Error('Account seed is missing')
```

File: yarn.lock [hunk 3]
```diff
@@ -2922,6 +2922,15 @@ cli-table3@^0.6.2:
   optionalDependencies:
```
Comment (rflechtner) [inline yarn.lock:2922]: can be deduped, run `npx yarn-dedupe yarn.lock` & `yarn install`

**Review by rflechtner — APPROVED** [review]
Love it! Not that it really matters, but maybe we can tweak the test.ts script command line interface a bit

File: yarn.lock (hunk 3 above)
Comment (weichweich) [reply yarn.lock:2922]: [84e291a](https://github.com/KILTprotocol/docs/pull/197/commits/84e291a2b3d4d00a765a73c58488ef042154bfc0) Didn't do much?

File: code_examples/sdk_examples/src/test.ts (hunk 2 above)
Comment (weichweich) [reply code_examples/sdk_examples/src/test.ts]: [167f846](https://github.com/KILTprotocol/docs/pull/197/commits/167f84609ed23dc27a287ab6baa3392317ec952d)

File: code_examples/sdk_examples/src/test.ts (hunk 1 above)
Comment (weichweich) [reply code_examples/sdk_examples/src/test.ts:29]: Sorry but i think I'm not doing this. Already spent too much time on this and it's not such a huge issue I guess?

Comment (rflechtner) [reply code_examples/sdk_examples/src/test.ts:29]: yeah just an idea

File: yarn.lock (hunk 3 above)
Comment (rflechtner) [reply yarn.lock:2922]: but it did something!

**Pull request merged**